renku/core/models/provenance/datasets.py

```diff
@@ -128,6 +152,25 @@ def _extract_identifier(self):
         parsed_url = urlparse(self.id)
         return list(Path(parsed_url.path).parents)[-3].name
 
+    def to_dataset_file(self, client, revision="HEAD"):
+        """Return an instance of renku.core.models.datasets.DatasetFile at a revision."""
+        from renku.core.models.datasets import DatasetFile
+
+        try:
+            return DatasetFile.from_revision(
+                client=client,
+                revision=revision,
+                added=self.date_added,
+                based_on=self.based_on,
+                external=self.is_external,
+                id=None,
+                path=self.entity.path,
+                source=self.source,
+                url=None,
+            )
+        except KeyError:  # path does not exists at revision
```

Comment:
Could we add a tag `NOTE:` and also which line from `.from_revision` is expected to raise this?

Reply:
I've changed the wording to make it clear which part of that function raises this error.